ibis/backends/dask/execution/generic.py

```diff
@@ -145,17 +145,19 @@
 def execute_arbitrary_series_mask(op, data, mask, aggcontext=None, **kwargs):
     """
     Note: we cannot use the pandas version because Dask does not support .iloc
+    See https://docs.dask.org/en/latest/dataframe-indexing.html. .loc will
+    only work if our index lines up with the label.
     """
+    data = data[mask] if mask is not None else data
     if op.how == 'first':
         index = 0
     elif op.how == 'last':
-        index = -1
+        index = len(data) - 1  # TODO - computation
```

Comment:
How is groupby handled here?

Reply:
Also, can you create an Jira to track this issue?

Reply:
The documentation says "Dask DataFrame does not track the length of partitions, making positional indexing with .iloc inefficient for selecting rows"

Does that mean here we trigger two pass of computation, one pass to compute the length, the second pass to index the row given the length? (I'd imagine this would be a full scan given what the documentation is describing)

Reply:
> How is groupby handled here?

I don't understand the question - this operation is only registered for regular `dd.Series`. 

> Also, can you create an Jira to track this issue?

Which issue?

Reply:
Oh sorry I mean the "TODO" here

Reply:
> I don't understand the question - this operation is only registered for regular dd.Series.

Oh I see.  Where is `table.groupby('key').first()` and `table.groupby('key').first()` handled?

Reply:
(It's fine if it's not, just want to make sure I didn't miss sth)

Reply:
Huh - for some reason that's in `numeric` via `execute_arbitrary_series_groupby`. 

Not sure why I put that there originally - I'll move it.

Reply:
Aha ok.